python/lib/dcm2bids_imaging_pipeline_lib/base_pipeline.py

```diff
@@ -78,109 +75,127 @@ def __init__(self, loris_getopt_obj, script_name):
         the script.
         """
 
         upload_id = self.options_dict["upload_id"]["value"]
         tarchive_path = self.options_dict["tarchive_path"]["value"] \
             if "tarchive_path" in self.options_dict.keys() else None
-        success = False
+
         if upload_id and tarchive_path:
-            self.imaging_upload_obj.create_imaging_upload_dict_from_upload_id(upload_id)
-            if not self.imaging_upload_obj.imaging_upload_dict:
+            mri_upload = try_get_mri_upload_with_id(self.env.db, upload_id)
+            if mri_upload is None:
                 log_error_exit(
                     self.env,
                     f"Did not find an entry in mri_upload associated with \'UploadID\' {upload_id}.",
                     lib.exitcode.SELECT_FAILURE,
                 )
-            tarchive_id = self.imaging_upload_obj.imaging_upload_dict["TarchiveID"]
-            if not tarchive_id:
+
+            self.mri_upload = mri_upload
+
+            if self.mri_upload.dicom_archive is None:
                 log_error_exit(
                     self.env,
                     f"UploadID {upload_id} is not linked to any tarchive in mri_upload.",
                     lib.exitcode.SELECT_FAILURE,
                 )
 
-            self.dicom_archive = try_get_dicom_archive_with_id(self.env.db, tarchive_id)
+            self.dicom_archive = self.mri_upload.dicom_archive
             if os.path.join(self.data_dir, 'tarchive', self.dicom_archive.archive_location) != tarchive_path:
                 log_error_exit(
                     self.env,
                     f"UploadID {upload_id} and ArchiveLocation {tarchive_path} do not refer to the same upload",
                     lib.exitcode.SELECT_FAILURE,
                 )
 
-        err_msg = ''
         if upload_id:
-            self.imaging_upload_obj.create_imaging_upload_dict_from_upload_id(upload_id)
-            if not self.imaging_upload_obj.imaging_upload_dict:
-                err_msg += f"Did not find an entry in mri_upload associated with 'UploadID' {upload_id}"
-            else:
-                if self.imaging_upload_obj.imaging_upload_dict["TarchiveID"]:
-                    tarchive_id = self.imaging_upload_obj.imaging_upload_dict["TarchiveID"]
-                    self.dicom_archive = try_get_dicom_archive_with_id(self.env.db, tarchive_id)
-                    if self.dicom_archive is not None:
-                        success = True
-                    else:
-                        err_msg += f"Could not load tarchive dictionary for TarchiveID {tarchive_id}"
+            mri_upload = try_get_mri_upload_with_id(self.env.db, upload_id)
+            if mri_upload is None:
+                log_error_exit(
+                    self.env,
+                    f"Did not find an entry in mri_upload associated with 'UploadID' {upload_id}",
+                    lib.exitcode.SELECT_FAILURE,
+                )
+
+            self.mri_upload = mri_upload
```

Comment:
Couldn't this be simplified this way? Is there a reason `self.mri_upload` is not assigned right away?

```suggestion
            self.mri_upload = try_get_mri_upload_with_id(self.env.db, upload_id)
            if self.mri_upload is None:
                log_error_exit(
                    self.env,
                    f"Did not find an entry in mri_upload associated with 'UploadID' {upload_id}",
                    lib.exitcode.SELECT_FAILURE,
                )
```